### PR TITLE
Update automation tests for the AWS gems to reduce the run time

### PR DESCRIPTION
**Pull request by junbo75**
- Refactor the automation tests code to significantly reduce the times for CDK deployment and running game launcher
- Run time-consuming verification simultaneously on separate threads
- Bug fixes for the CDK fixture
- Regenerate the test level for Client Auth gem
- Automation tests passed on Jenkins: https://jenkins-pipeline.agscollab.com/job/O3DE-LY-FORK/job/LYN-4296/10/:

16:30:20        Start 46: AutomatedTesting::AWSTests.periodic::TEST_RUN
16:51:13  19/21 Test #46: AutomatedTesting::AWSTests.periodic::TEST_RUN ....................................   Passed  1252.33 sec

Need to follow up with infra since it took more than 15 minutes for deploying and destroying AWS resources but we have 25 minutes in total for all our automation tests. Ideally we should have the resource deployment run via separate Jenkins job as Hrudesh proposed at https://wiki.agscollab.com/display/lmbr/AWSI+Automation

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/AWS/Windows/core/test_aws_resource_interaction.py [hunk 1]
```diff
@@ -57,130 +88,57 @@ def setup(launcher: pytest.fixture, cdk: pytest.fixture, resource_mappings: pyte
...
+         '(Script) - [S3] Head object success: Object example.txt is found.',
+         '(Script) - [S3] Get object success: Object example.txt is downloaded.',
+         '(Script) - [Lambda] Completed Invoke',
+         '(Script) - [Lambda] Invoke success: {"statusCode": 200, "body": {}}',
+         '(Script) - [DynamoDB] Results finished']])
+    @pytest.mark.parametrize('unexpected_lines', [
+        ['(Script) - [S3] Head object error: No response body.',
+         '(Script) - [S3] Get object error: Request validation failed, output file directory doesn\'t exist.',
+         '(Script) - Request validation failed, output file miss full path.',
+         '(Script) - ']])
```
Comment (onecent1101) [inline AutomatedTesting/Gem/PythonTests/AWS/Windows/core/test_aws_resource_interaction.py:110]: the only concern is how to know which scriptcanvas fails if error happens?

Comment (junbo75) [reply AutomatedTesting/Gem/PythonTests/AWS/Windows/core/test_aws_resource_interaction.py:110]: Log monitor will raise an error with the unexpected line it found in the log.

**Review by AMZN-hdoke — APPROVED** [review]

**Review by onecent1101 — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/AWS/Windows/aws_metrics/aws_metrics_utils.py [hunk 2]
```diff
@@ -7,12 +7,12 @@
 import logging
 import pathlib
 import pytest
+import time
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/AWS/Windows/aws_metrics/aws_metrics_utils.py:10]: nit: does not appear to be used in modified code

File: AutomatedTesting/Gem/PythonTests/AWS/Windows/cdk/cdk_utils.py [hunk 3]
```diff
@@ -71,15 +71,14 @@ def __init__(self):
...
     def setup(self, cdk_path: str, project: str, account_id: str,
-                 workspace: pytest.fixture, session: boto3.session.Session, bootstrap_required: bool):
+              workspace: pytest.fixture, session: boto3.session.Session):
         """
         :param cdk_path: Path where cdk app.py is stored.
         :param project: Project name used for cdk project name env variable.
         :param account_id: AWS account id to use with cdk application.
         :param workspace: ly_test_tools workspace fixture.
-        :param workspace: bootstrap_required deploys bootstrap stack.
+        :param session: Session for interacting with AWS.
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/AWS/Windows/cdk/cdk_utils.py:80]: nit: Current boto3 session, provides credentials and region?

File: AutomatedTesting/Gem/PythonTests/AWS/Windows/core/test_aws_resource_interaction.py [hunk 4]
```diff
@@ -57,130 +88,57 @@ def setup(launcher: pytest.fixture, cdk: pytest.fixture, resource_mappings: pyte
...
                                 aws_utils: pytest.fixture,
-                                ):
+                                expected_lines: typing.List[str],
+                                unexpected_lines: typing.List[str]):
         """
-        Setup: Deploys  the CDK application
-        Test: Runs a launcher with a level that loads a scriptcanvas that pulls a DynamoDB table value.
-        Verification: The value is output in the logs and verified by the test.
+        Setup: Deploys cdk and updates resource mapping file.
+        Tests: Getting AWS credentials for no signed in user.
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/AWS/Windows/core/test_aws_resource_interaction.py:122]: nit: "no signed in user"? Maybe "Getting credentials when no credentials are configured"

**Review by lmbr-pip — APPROVED** [review]
LG

**Pull request merged**